### PR TITLE
wip: automate json schema definition of some config structs

### PR DESCRIPTION
**Pull request by sreis**
Cross-linking to #580.

Uses github.com/alecthomas/jsonschema to convert golang structs that define Cortex, Loki and Alertmanager configuration to JSON schema. The JSON schema will be used to validate user input.

Currently only prints stuff to stdout.

In the process learned that Loki does not support golang modules properly and we need to pin to specific commits with `go get github.com/grafana/loki@COMMIT_HASH`.

### CODE REVIEW

File: go/tools/cortex-jsonschema-exporter/alertmanager-config.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+	"github.com/alecthomas/jsonschema"
+	"github.com/cortexproject/cortex/pkg/alertmanager"
+)
+
+//
+// Prints the json schema definition of the Cortex configuration to stdout.
+//
+func main() {
+	schema := jsonschema.Reflect(&alertmanager.Config{})
```
Comment (sreis) [inline go/tools/cortex-jsonschema-exporter/alertmanager-config.go:29]: @terrcin Can you tell me which struct you looked at when you created the schema in https://github.com/opstrace/opstrace/blob/main/packages/app/src/client/validation/alertmanager/config/types.ts ?

Comment (terrcin) [reply go/tools/cortex-jsonschema-exporter/alertmanager-config.go:29]: @sreis I looked at the alertmanager config documentation itself and manually created those types based on that. Did it at the same time as the Yup validations.

Comment (sreis) [reply go/tools/cortex-jsonschema-exporter/alertmanager-config.go:29]: Ok. I'll try to find the corresponding golang struct.

Comment (sreis) [reply go/tools/cortex-jsonschema-exporter/alertmanager-config.go]: Changed to use the one that looked the most similar. 

@terrcin Any ideias how we can compare the autogenerated JSON schema with https://github.com/opstrace/opstrace/blob/main/packages/app/src/client/validation/alertmanager/config/schema.json ?

Comment (terrcin) [reply go/tools/cortex-jsonschema-exporter/alertmanager-config.go]: @sreis maybe by eye at first. Do you have an example of the output somewhere I can look it? Also could do over zoom when we're next both online.

**Comment by terrcin** [comment]
@sreis I've taken a quick look at the generated alertmanager json schema compared to the one I manually created. The main differences are the generated schema additionally specifies which fields are required while also being an older version of the spec format "draft-04" vs "draft-07".

**Comment by sreis** [comment]
Closing since there's ongoing work to support runtime config API.

**Pull request closed without merging**